### PR TITLE
WIP - Track resources that need retrying in namespace cleanup

### PR DESCRIPTION
**Pull request by liggitt**
Builds on https://github.com/kubernetes/kubernetes/pull/47434

### CODE REVIEW

File: pkg/controller/namespace/deletion/namespaced_resources_deleter.go [hunk 1]
```diff
@@ -38,7 +38,11 @@ import (
 
 // Interface to delete a namespace with all resources in it.
 type NamespacedResourcesDeleterInterface interface {
-	Delete(nsName string) error
+	// Delete deletes the specified resources inside the specified namespace.
+	// If gvrsToDelete is nil, deletes all resources.
```
Comment (smarterclayton) [inline pkg/controller/namespace/deletion/namespaced_resources_deleter.go]: empty

File: pkg/controller/namespace/namespace_controller.go [hunk 2]
```diff
@@ -58,6 +60,23 @@ type NamespaceController struct {
 	queue workqueue.RateLimitingInterface
 	// helper to delete all resources in the namespace when the namespace is deleted.
 	namespacedResourcesDeleter deletion.NamespacedResourcesDeleterInterface
+
+	gvrsToDeleteLock sync.Mutex
+	gvrsToDelete     map[string][]schema.GroupVersionResource
```
Comment (smarterclayton) [inline pkg/controller/namespace/namespace_controller.go]: Yuck, why can't this be inside namespacedResourcesDeleter?

Comment (liggitt) [inline pkg/controller/namespace/namespace_controller.go]: quick and dirty, mostly just an experiment

**Pull request closed without merging**